### PR TITLE
Fix Null Pointer Exception thrown when resolving paths in ProjectPaths utils

### LINKED ISSUE #28744: [LS] NPE at startup
Get the following error in VSCode, 

```
'bal' command is picked up from the plugin.
Plugin version: 2.0.0
Ballerina version: Swan Lake Alpha 2
Using Ballerina CLI command 'bal.bat' for Language server.
[Error - 11:30:22 AM] Operation 'text/didOpen' failed! {uri: '/d%3A/test-pack/tmp.bal', error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at io.ballerina.projects.util.ProjectPaths.isNonDefaultModuleTestFile(ProjectPaths.java:177)
	at io.ballerina.projects.util.ProjectPaths.isStandaloneBalFile(ProjectPaths.java:146)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectKindAndProjectRoot(BallerinaWorkspaceManager.java:425)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectRoot(BallerinaWorkspaceManager.java:421)
	at java.base/java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:330)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectRoot(BallerinaWorkspaceManager.java:115)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.didOpen(BallerinaWorkspaceManager.java:224)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.didOpen(BallerinaTextDocumentService.java:506)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

[Error - 11:30:22 AM] Operation 'text/codeAction' failed! {uri: '/d%3A/test-pack/tmp.bal', [1:1]- [1:1], error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at io.ballerina.projects.util.ProjectPaths.isNonDefaultModuleTestFile(ProjectPaths.java:177)
	at io.ballerina.projects.util.ProjectPaths.isStandaloneBalFile(ProjectPaths.java:146)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectKindAndProjectRoot(BallerinaWorkspaceManager.java:425)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectRoot(BallerinaWorkspaceManager.java:421)
	at java.base/java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:330)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectRoot(BallerinaWorkspaceManager.java:115)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectPair(BallerinaWorkspaceManager.java:433)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.project(BallerinaWorkspaceManager.java:126)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.document(BallerinaWorkspaceManager.java:156)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.syntaxTree(BallerinaWorkspaceManager.java:168)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:58)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:133)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:329)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

[Error - 11:30:22 AM] Operation 'text/foldingRange' failed! {uri: '/d%3A/test-pack/tmp.bal', error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at io.ballerina.projects.util.ProjectPaths.isNonDefaultModuleTestFile(ProjectPaths.java:177)
	at io.ballerina.projects.util.ProjectPaths.isStandaloneBalFile(ProjectPaths.java:146)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectKindAndProjectRoot(BallerinaWorkspaceManager.java:425)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectRoot(BallerinaWorkspaceManager.java:421)
	at java.base/java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:330)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectRoot(BallerinaWorkspaceManager.java:115)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectPair(BallerinaWorkspaceManager.java:433)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.project(BallerinaWorkspaceManager.java:126)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.document(BallerinaWorkspaceManager.java:156)
	at org.ballerinalang.langserver.foldingrange.FoldingRangeProvider.getSyntaxTree(FoldingRangeProvider.java:66)
	at org.ballerinalang.langserver.foldingrange.FoldingRangeProvider.getFoldingRange(FoldingRangeProvider.java:45)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$foldingRange$16(BallerinaTextDocumentService.java:573)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

[Error - 11:30:22 AM] Operation 'text/codeLens' failed! {uri: '/d%3A/test-pack/tmp.bal', error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at io.ballerina.projects.util.ProjectPaths.isNonDefaultModuleTestFile(ProjectPaths.java:177)
	at io.ballerina.projects.util.ProjectPaths.isStandaloneBalFile(ProjectPaths.java:146)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectKindAndProjectRoot(BallerinaWorkspaceManager.java:425)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectRoot(BallerinaWorkspaceManager.java:421)
	at java.base/java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:330)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectRoot(BallerinaWorkspaceManager.java:115)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectPair(BallerinaWorkspaceManager.java:433)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.project(BallerinaWorkspaceManager.java:126)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.document(BallerinaWorkspaceManager.java:156)
	at org.ballerinalang.langserver.codelenses.providers.DocsCodeLensesProvider.getLenses(DocsCodeLensesProvider.java:69)
	at org.ballerinalang.langserver.codelenses.CodeLensUtil.getCodeLenses(CodeLensUtil.java:48)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeLens$9(BallerinaTextDocumentService.java:367)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

[Error - 11:30:22 AM] Operation 'text/codeLens' failed! {uri: '/d%3A/test-pack/tmp.bal', error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at io.ballerina.projects.util.ProjectPaths.isNonDefaultModuleTestFile(ProjectPaths.java:177)
	at io.ballerina.projects.util.ProjectPaths.isStandaloneBalFile(ProjectPaths.java:146)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectKindAndProjectRoot(BallerinaWorkspaceManager.java:425)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectRoot(BallerinaWorkspaceManager.java:421)
	at java.base/java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:330)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectRoot(BallerinaWorkspaceManager.java:115)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectPair(BallerinaWorkspaceManager.java:433)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.project(BallerinaWorkspaceManager.java:126)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.document(BallerinaWorkspaceManager.java:156)
	at org.ballerinalang.langserver.codelenses.providers.DocsCodeLensesProvider.getLenses(DocsCodeLensesProvider.java:69)
	at org.ballerinalang.langserver.codelenses.CodeLensUtil.getCodeLenses(CodeLensUtil.java:48)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeLens$9(BallerinaTextDocumentService.java:367)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

[Error - 11:30:22 AM] Operation 'text/codeLens' failed! {uri: '/d%3A/test-pack/tmp.bal', error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at io.ballerina.projects.util.ProjectPaths.isNonDefaultModuleTestFile(ProjectPaths.java:177)
	at io.ballerina.projects.util.ProjectPaths.isStandaloneBalFile(ProjectPaths.java:146)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectKindAndProjectRoot(BallerinaWorkspaceManager.java:425)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.computeProjectRoot(BallerinaWorkspaceManager.java:421)
	at java.base/java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:330)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectRoot(BallerinaWorkspaceManager.java:115)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.projectPair(BallerinaWorkspaceManager.java:433)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.project(BallerinaWorkspaceManager.java:126)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.document(BallerinaWorkspaceManager.java:156)
	at org.ballerinalang.langserver.codelenses.providers.DocsCodeLensesProvider.getLenses(DocsCodeLensesProvider.java:69)
	at org.ballerinalang.langserver.codelenses.CodeLensUtil.getCodeLenses(CodeLensUtil.java:48)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeLens$9(BallerinaTextDocumentService.java:367)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

### PR DESCRIPTION
**Pull request by azinneera**
Fixes #28744

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectPaths.java [hunk 1]
```diff
@@ -53,6 +53,11 @@ public static Path packageRoot(Path filepath) throws ProjectException {
             throw new ProjectException("provided path is not a regular file");
         }
 
+        // Check if the file is inside a Ballerina package directory
+        if (findProjectRoot(filepath).isEmpty()) {
+            throw new ProjectException("provided file path does not belong to a Ballerina package");
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectPaths.java:58]: Wouldn't it be better if we include the file path that the project API was trying to lookup in this error message?

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectPaths.java]: Addressed in c4a17d7

File: .github/workflows/CI_ubuntu_windows.yml [hunk 2]
```diff
@@ -75,5 +75,5 @@ jobs:
           restore-keys: ${{ runner.os }}-gradle
 
       - name: Build with Gradle
-        run: ./gradlew.bat build --continue -x :ballerina-lang:test -x :jballerina-unit-test:test -x :jballerina-integration-test:test -x :jballerina-debugger-integration-test:test  -x :ballerina-cli:test -x :ballerina-shell:shell-cli:test -x createJavadoc --stacktrace -scan --console=plain --no-daemon --no-parallel
+        run: ./gradlew.bat build --continue -x :jballerina-unit-test:test -x :jballerina-integration-test:test -x :jballerina-debugger-integration-test:test  -x :ballerina-cli:test -x :ballerina-shell:shell-cli:test -x createJavadoc --stacktrace -scan --console=plain --no-daemon --no-parallel
```
Comment (hevayo) [inline .github/workflows/CI_ubuntu_windows.yml:78]: We need to remove cli tests also from this list. can handle from a separate PR

**Review by hevayo — APPROVED** [review]

File: .github/workflows/CI_ubuntu_windows.yml (hunk 2 above)
Comment (azinneera) [reply .github/workflows/CI_ubuntu_windows.yml:78]: It was disabled due to an intermittent failure when cleaning the target. It is tracked by https://github.com/ballerina-platform/ballerina-lang/issues/27455

**Pull request merged**